devito/dse/extended_sympy.py

```diff
@@ -41,7 +41,16 @@ def xreplace(self, rule):
 
 
 class Eq(sympy.Eq, FrozenExpr):
-    pass
+
+    """
+    A customized version of :class:`sympy.Eq` which suppresses
+    evaluation.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        kwargs['evaluate'] = False
+        obj = sympy.Eq.__new__(cls, *args, **kwargs)
```

Comment:
Just of out curiosity, where did it break w/o this ?

Reply:
The reason is subtle and only applies for a few corner-cases in `test_operator.py`; effectively degenerate expressions like `Eq(a, a + 3)` will now immediately evaluate to `False`, so we need to suppress evaluation.

Reply:
Add `Eq` to `_all_` and should be good